tests/test_gpu.py

```diff
@@ -210,19 +210,24 @@ def test_gpu_metadata_total_power(self):
         from codecarbon.core.units import Energy, Power, Time
         from codecarbon.external.hardware import GPU
 
-        gpu1_energy1 = Energy.from_millijoules(149701)
-        gpu1_energy2 = Energy.from_millijoules(180000)
-        gpu2_energy1 = Energy.from_millijoules(149702)
-        gpu2_energy2 = Energy.from_millijoules(180000)
-
-        gpu2_power = Power.from_energies_and_delay(gpu1_energy1, gpu1_energy2, Time(5))
-        gpu1_power = Power.from_energies_and_delay(gpu2_energy1, gpu2_energy2, Time(5))
-        expected_power = gpu1_power + gpu2_power
+        energy_consumption = [100_701, 149_702, 180_001, 180_002, 190_001, 200_002]
+        # gpu1_energy1 = Energy.from_millijoules(energy_consumption[0])
+        gpu1_energy2 = Energy.from_millijoules(energy_consumption[2])
+        gpu1_energy3 = Energy.from_millijoules(energy_consumption[4])
+        # gpu2_energy1 = Energy.from_millijoules(energy_consumption[1])
+        gpu2_energy2 = Energy.from_millijoules(energy_consumption[3])
+        gpu2_energy3 = Energy.from_millijoules(energy_consumption[5])
+
+        # gpu2_power = Power.from_energies_and_delay(gpu1_energy1, gpu1_energy2, Time(5))
+        gpu2_power2 = Power.from_energies_and_delay(gpu1_energy2, gpu1_energy3, Time(5))
+        # gpu1_power = Power.from_energies_and_delay(gpu2_energy1, gpu2_energy2, Time(5))
+        gpu1_power2 = Power.from_energies_and_delay(gpu2_energy2, gpu2_energy3, Time(5))
+        expected_power = gpu1_power2 + gpu2_power2
```

Comment:
I need to do this because it seems the first calls to `nvmlDeviceGetTotalEnergyConsumption` are lost because of the new code of hardware.py:from_utils that made two calls to the class :
Previous code: `        return cls(gpu_ids=gpu_ids)`

New code:
```py
        gpus = cls(gpu_ids=gpu_ids)
        new_gpu_ids = gpus._get_gpu_ids()
        if len(new_gpu_ids) < gpus.num_gpus:
            logger.warning(
                f"You have {gpus.num_gpus} GPUs but we will monitor only {len(gpu_ids)} of them. Check your configuration."
            )
        return cls(gpu_ids=new_gpu_ids)
```